backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -289,7 +291,42 @@ object Compiler {
       // Sources are all files
       val sources = inputs.sources.map(path => converter.toVirtualFile(path.underlying))
       val classpath = inputs.classpath.map(path => converter.toVirtualFile(path.underlying))
-      val optionsWithoutFatalWarnings = inputs.scalacOptions.flatMap { option =>
+      def existsReleaseSetting = inputs.scalacOptions.exists(opt =>
+        opt.startsWith("-release") ||
+          opt.startsWith("--release") ||
+          opt.startsWith("-java-output-version")
+      )
+      def sameHome = inputs.javacBin match {
+        case Some(bin) => bin.getParent.getParent == JavaRuntime.home
+        case None => false
+      }
+
+      val scalacOptions = inputs.javacBin.flatMap(binary =>
+        // <JAVA_HOME>/bin/java
+        JavaRuntime.getJavaVersionFromJavaHome(binary.getParent.getParent)
+      ) match {
+        case None => inputs.scalacOptions
+        case Some(_) if existsReleaseSetting || sameHome => inputs.scalacOptions
+        case Some(version) =>
+          try {
+            val numVer = if (version.startsWith("1.8")) 8 else version.takeWhile(_.isDigit).toInt
+            val bloopNumVer = JavaRuntime.version.takeWhile(_.isDigit).toInt
+            if (bloopNumVer > numVer) {
+              inputs.scalacOptions ++ List("-release", numVer.toString())
+            } else {
+              logger.warn(
+                s"Bloop is run with ${JavaRuntime.version} but your code requires $version to compile, " +
```

Comment:
```suggestion
                s"Bloop is running with ${JavaRuntime.version} but your code requires $version to compile, " +
```
Should we maybe change it to `running` to hint at the fact that it needs to stop and be restarted with that version if necessary?

Reply:
Done!